src/index/diskann/diskann_config.h

```diff
@@ -148,6 +172,10 @@ class DiskANNConfig : public BaseConfig {
                 if (!search_list_size.has_value()) {
                     search_list_size = kDefaultSearchListSizeForBuild;
                 }
+                pq_code_budget_gb =
+                    std::max(pq_code_budget_gb.value(), pq_code_budget_gb_ratio.value() * vec_field_size_gb.value());
+                search_cache_budget_gb = std::max(search_cache_budget_gb.value(),
+                                                  search_cache_budget_gb_ratio.value() * vec_field_size_gb.value());
```

Comment:
The purpose is replacement. Instead of `max`, let's directly use `ratio` once it is provided.

Reply:
The main consideration here is compatibility in distributed scenarios. For example, if the parameter-filling code is old, it won't have pq_code_budget_gb_ratio/search_cache_budget_gb_ratio

Reply:
then you can pick the original one

Reply:
this will be fixed after https://github.com/milvus-io/milvus/pull/36844 merged. A more clear logic will be provided (choose ratio parameter with high priority)